File: src/main/java/Model/ManualReview.java
package Model;

import java.sql.Timestamp;

/**
 * Representación de la tabla 'ManualReview' en la base de datos.
 * 
 * CREATE TABLE ManualReview (
 *    reviewId INT AUTO_INCREMENT PRIMARY KEY,
 *    submissionId INT,                      -- Referencia a la tarea entregada
 *    id_professor VARCHAR(50),              -- Referencia al profesor que revisó
 *    grade DECIMAL(5,2),                    -- Calificación manual
 *    comments TEXT,                          -- Comentarios manuales
 *    reviewDate TIMESTAMP DEFAULT CURRENT_TIMESTAMP, -- Fecha de la revisión
 *    FOREIGN KEY (submissionId) REFERENCES Submission(submissionId) ON DELETE CASCADE,
 *    FOREIGN KEY (id_professor) REFERENCES User(id_user) ON DELETE CASCADE
 * );
 * 
 * @author kathe
 */
public class ManualReview {
    private int reviewId;
    private Submission submissionId;   
    private User id_professor;  
    private double grade;   
    private String comments;            // Comentarios del profesor
    private Timestamp reviewDate;       // Fecha de la revisión

    // Constructor
    public ManualReview(int reviewId, Submission submissionId, User id_professor, double grade, String comments, Timestamp reviewDate) {
        this.reviewId = reviewId;
        this.submissionId = submissionId;
        this.id_professor = id_professor;
        this.grade = grade;
        this.comments = comments;
        this.reviewDate = reviewDate;
    }

    // Getters y Setters
    public int getReviewId() {
        return reviewId;
    }

    public void setReviewId(int reviewId) {
        this.reviewId = reviewId;
    }

    public Submission getSubmissionId() {
        return submissionId;
    }

    public void setSubmissionId(Submission submissionId) {
        this.submissionId = submissionId;
    }

    public User getId_professor() {
        return id_professor;
    }

    public void setId_professor(User id_professor) {
        this.id_professor = id_professor;
    }

    public double getGrade() {
        return grade;
    }

    public void setGrade(double grade) {
        this.grade = grade;
    }

    public String getComments() {
        return comments;
    }

    public void setComments(String comments) {
        this.comments = comments;
    }

    public Timestamp getReviewDate() {
        return reviewDate;
    }

    public void setReviewDate(Timestamp reviewDate) {
        this.reviewDate = reviewDate;
    }

    // Método toString() para representar el objeto en formato legible
    @Override
    public String toString() {
        return "ManualReview{" +
                "reviewId=" + reviewId +
                ", submissionId=" + submissionId +
                ", id_professor=" + id_professor +
                ", grade=" + grade +
                ", comments='" + comments + '\'' +
                ", reviewDate=" + reviewDate +
                '}';
    }
}

File: flask-mvc-starter-kit/app/models/Manual_Review.py
from flask_sqlalchemy import SQLAlchemy
from app import db

class ManualReview(db.Model):
    __tablename__ = 'ManualReview'
    
    reviewId = db.Column(db.Integer, primary_key=True, autoincrement=True)
    submissionId = db.Column(db.Integer, db.ForeignKey('Submission.submissionId', ondelete="CASCADE"), nullable=False)
    id_professor = db.Column(db.String(50), db.ForeignKey('User.id_user', ondelete="CASCADE"), nullable=False)
    grade = db.Column(db.Numeric(5, 2), nullable=False)
    comments = db.Column(db.Text, nullable=True)
    reviewDate = db.Column(db.DateTime, default=db.func.current_timestamp())

    # Relaciones
    submission = db.relationship("Submission", backref=db.backref("manual_reviews", lazy=True))
    professor = db.relationship("User", backref=db.backref("manual_reviews", lazy=True))

    def to_dict(self):
        return {
            "reviewId": self.reviewId,
            "submissionId": self.submissionId,
            "id_professor": self.id_professor,
            "grade": str(self.grade),  # Convertir a string para JSON
            "comments": self.comments,
            "reviewDate": self.reviewDate,
            "submission": self.submission.to_dict() if self.submission else None,
            "professor": self.professor.to_dict() if self.professor else None
        }